### DISCUSSION TITLE
Compile scripts to make them portable

### DISCUSSION THREAD

**Discussion by balupton** [nushell/nushell#6494] [Ideas]
Would be great to be able to compile nushell scripts such that I get a portable binary that works independently and can be used standalone with whatever shell.

If this lands, I'll probably start rewriting https://github.com/bevry/dorothy in nushell as the biggest issue with it currently is that the scripts/commands depend on each other and thus aren't portable. Would be much nicer if people could just download the commands they want, on whatever setup they have, rather than needing everything.

**Comment by kubouch** [comment]
Compiled scripts would definitely be super cool but we already have a lot on our roadmap and currently don't plan this for 1.0, unless someone enthusiastic shows up and starts hacking on it. Maybe the least-effort way would be to generate Rust code from Nushell and then compile that. That sounds like it should be reasonably doable.

Also, at some point, we plan to have some solution to package management. This is a planned feature for 1.0 but we haven't started it yet. There is an unofficial package manager being developed, though, which might be usable for some use cases: https://github.com/skelly37/nupac.
